x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/InferenceIngestIT.java

```diff
@@ -0,0 +1,530 @@
+                is(true));
+        }
+
+        assertThat(client().admin().cluster().preparePutPipeline("simple_classification_pipeline",
+            new BytesArray(CLASSIFICATION_PIPELINE.getBytes(StandardCharsets.UTF_8)),
+            XContentType.JSON).get().isAcknowledged(), is(true));
+
+        assertThat(client().admin().cluster().preparePutPipeline("simple_regression_pipeline",
+            new BytesArray(REGRESSION_PIPELINE.getBytes(StandardCharsets.UTF_8)),
+            XContentType.JSON).get().isAcknowledged(), is(true));
+
+        for (int i = 0; i < 10; i++) {
+            client().prepareIndex("index_for_inference_test", "_doc")
+                .setSource(new HashMap<>(){{
+                    put("col1", randomFrom("female", "male"));
+                    put("col2", randomFrom("S", "M", "L", "XL"));
+                    put("col3", randomFrom("true", "false", "none", "other"));
+                    put("col4", randomIntBetween(0, 10));
+                }})
+                .setPipeline("simple_classification_pipeline")
+                .get();
+
+            client().prepareIndex("index_for_inference_test", "_doc")
+                .setSource(new HashMap<>(){{
+                    put("col1", randomFrom("female", "male"));
+                    put("col2", randomFrom("S", "M", "L", "XL"));
+                    put("col3", randomFrom("true", "false", "none", "other"));
+                    put("col4", randomIntBetween(0, 10));
+                }})
+                .setPipeline("simple_regression_pipeline")
+                .get();
+        }
+
+        assertThat(client().admin().cluster().prepareDeletePipeline("simple_classification_pipeline").get().isAcknowledged(),
+            is(true));
+
+        assertThat(client().admin().cluster().prepareDeletePipeline("simple_regression_pipeline").get().isAcknowledged(),
+            is(true));
+
+        client().admin().indices().refresh(new RefreshRequest("index_for_inference_test")).get();
+
+        assertThat(client().search(new SearchRequest().indices("index_for_inference_test")
+                .source(new SearchSourceBuilder()
+                    .size(0)
+                    .trackTotalHits(true)
+                    .query(QueryBuilders.boolQuery()
+                        .filter(QueryBuilders.existsQuery("regression_value"))))).get().getHits().getTotalHits().value,
+            equalTo(20L));
+
+        assertThat(client().search(new SearchRequest().indices("index_for_inference_test")
+                .source(new SearchSourceBuilder()
+                    .size(0)
+                    .trackTotalHits(true)
+                    .query(QueryBuilders.boolQuery()
+                        .filter(QueryBuilders.existsQuery("result_class"))))).get().getHits().getTotalHits().value,
+            equalTo(20L));
+
+    }
+
+    public void testSimulate() {
+        String source = "{\n" +
```

Comment:
[non-actionable] Waiting for multiline raw string literals being introduced to Java...

Reply:
No joke!
```
"""
"""
```
cannot be added soon enough!